src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThread.java

```diff
@@ -314,4 +314,13 @@ public static CpsThread current() {
         // getExecution().getOwner() would be useful but seems problematic.
         return "Thread #" + id + String.format(" @%h", this);
     }
+
+    /**
+     * Gets a the timeout collect from a System property or default value.
+     * Use -Dorg.jenkinsci.plugins.workflow.cps.CpsThread.runTimeoutMinutes
+     * @return
+     */
+    public long getRunTimeoutMinutes() {
```

Comment:
Typically such tuning constants are `public static /* not final */` fields, for ease of manipulation from `/script` etc.